source/slang/slang-ir-spirv-legalize.cpp

```diff
@@ -254,8 +254,30 @@ struct SPIRVLegalizationContext : public SourceEmitterBase
             }
             else
             {
-                if (!as<IRDecoration>(use->getUser()))
+                auto user = use->getUser();
+                if (!as<IRDecoration>(user))
                 {
+                    // do not emit load if we have a `store(Ptr<T>, globalParam<T>)`, instead
+                    // replace the store(Ptr<T>, globalParam<T>) with a globalParam<T> since 
+                    // this is the end product anyways (save on: 1 load, 1 store op). 
+                    // This also stops TextureType from breaking
+                    if(auto opStore = as<IRStore>(user))
+                    {
+                        auto dstType = opStore->getOperand(0)->getDataType();
+                        auto srcType = opStore->getOperand(1)->getDataType();
+                        // Cover the case where we have a Ptr<Ptr<T>>
+                        while (dstType->getOp() == kIROp_PtrType && srcType->getOp() == kIROp_PtrType)
```

Comment:
I don't quite understand the logic here. Why do we need to handle the case of ptr<ptr<T>?

Reply:
I don't know if slang supports Ptr<Ptr< or plans to, it can be removed if not important.

Reply:
This currently will occur with pointer-typed user code, such as:
```
void main () {
   int* ptr;
}
```

will be lowered as:
```
%ptr = IRVar : Ptr<Ptr<int>>
```

But I don't know why this will come up in spirv legalization pass.

Reply:
I suppose I will leave out this change for now then (won't be hard to add if important anyways)